tools/elfdeps.cc

```diff
@@ -96,27 +90,31 @@ static const char *mkmarker(GElf_Ehdr *ehdr)
     return marker;
 }
 
-static void addDep(ARGV_t *deps,
-		   const char *soname, const char *ver, const char *marker)
+static void addDep(std::vector<std::string> & deps, const std::string & dep)
 {
-    char *dep = NULL;
+    deps.push_back(dep);
+}
 
-    if (skipSoname(soname))
+static void addSoDep(std::vector<std::string> & deps,
+		     const std::string & soname,
+		     const std::string & ver, const std::string & marker)
+{
+    if (skipSoname(soname.c_str()))
```

Comment:
This was probably forgotten as part of the "Use a C++ string for skipSoname() convenience" commit :sweat_smile:

Reply:
Oh, now I somewhat recall that you can actually assigned a C string to an `std::string` variable. Still, it seems like the conversion can be removed for clarity anyway.

Reply:
Oh yup, that's just an accidental leftove caused by "wrong" refactoring order - reordered the two relevant commits and voila the issue goes away :sweat_smile: Thanks for spotting.